library/billrep.inc

```diff
@@ -154,26 +154,27 @@ function getBillsListBetween(
     "order by billing.pid, billing.date ASC";
 
     $res = sqlStatement($sql, array($code_type));
-    $string = "( ";
+    $array = array();
     for ($iter=0; $row=sqlFetchArray($res); $iter++) {
-        $string .= $row{"id"}.",";
+        array_push($array, $row{"id"});
     }
 
-    $string = substr($string, 0, strlen($string)-1);
-    $string .= ")";
-    return $string;
+    return $array;
 }
 
-function billCodesList($list, $skip = "()")
+function billCodesList($list, $skip = [])
 {
-    if ($list == "()") {
+    if (empty($list)) {
         return;
     }
 
-    if ($skip == "()") {
-        sqlStatement("update billing set billed=1 where id in $list");
+    $sqlBindArray = array_merge($list, $skip);
+    $in = str_repeat('?,', count($list) - 1) . '?';
```

Comment:
You could use array diff to simplify your control flow logic and your sql. http://php.net/manual/fr/function.array-diff.php

Reply:
Good call.  Thanks!

Reply:
Agreed @alecharmon ; very nice call to simplify this code!